bundles/io/org.eclipse.smarthome.io.console/src/main/java/org/eclipse/smarthome/io/console/internal/extension/MetadataConsoleCommandExtension.java

```diff
@@ -91,21 +98,28 @@ public void execute(String[] args, Console console) {
         }
     }
 
-    private void listMetadata(Console console, String itemName, String namespace) {
+    private void listMetadata(Console console, String itemName, String namespace, boolean internal) {
         if (itemName == null) {
-            metadataRegistry.stream().map(Metadata::toString).forEach(console::println);
-        } else if (namespace == null) {
-            metadataRegistry.stream().filter(MetadataPredicates.ofItem(itemName)).map(Metadata::toString)
+            metadataRegistry.stream().filter(m -> isInternal(m, internal)).map(Metadata::toString)
                     .forEach(console::println);
+        } else if (namespace == null) {
+            metadataRegistry.stream().filter(MetadataPredicates.ofItem(itemName)).filter(m -> isInternal(m, internal))
+                    .map(Metadata::toString).forEach(console::println);
         } else {
             MetadataKey key = new MetadataKey(namespace, itemName);
-            Metadata metadata = metadataRegistry.get(key);
-            if (metadata != null) {
-                console.println(metadata.toString());
+            if (metadataRegistry.isInternalNamespace(namespace) == internal) {
+                Metadata metadata = metadataRegistry.get(key);
+                if (metadata != null) {
+                    console.println(metadata.toString());
+                }
             }
         }
     }
 
+    private boolean isInternal(@NonNull Metadata metadata, boolean internal) {
+        return metadataRegistry.isInternalNamespace(metadata.getUID().getNamespace()) == internal;
```

Comment:
While this is working - I'd suggest to extend the Metadata API in a way to have `MetadataRegistry#isInternalNamespace` to be static and having a method `isInternalNamespace` added to  `MetadataKey` and `Metadata` itself.